node/impl/full/eth.go

```diff
@@ -570,12 +570,68 @@ func (a *EthModule) EthGetTransactionReceiptLimited(ctx context.Context, txHash
 	return &receipt, nil
 }
 
-func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(context.Context, ethtypes.EthHash, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(ctx context.Context, blkHash ethtypes.EthHash, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	ts, err := a.Chain.GetTipSetByCid(ctx, blkHash.ToCid())
+	if err != nil {
+		if err == ErrNullRound {
+			return nil, xerrors.Errorf("block hash %s was a null round", blkHash)
+		}
+		return nil, xerrors.Errorf("failed to get tipset by cid: %w", err)
+	}
+
+	return a.getTransactionByTipsetAndIndex(ctx, ts, index)
+}
+
+func (a *EthAPI) EthGetTransactionByBlockNumberAndIndex(ctx context.Context, blkParam string, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	ts, err := getTipsetByBlockNumber(ctx, a.Chain, blkParam, true)
+	if err != nil {
+		if err == ErrNullRound {
+			return nil, xerrors.Errorf("block number %s was a null round", blkParam)
+		}
+		return nil, xerrors.Errorf("failed to get tipset for block %s: %w", blkParam, err)
```

Comment:
```suggestion
		return nil, xerrors.Errorf("failed to get tipset for block %s: %w", blkParam, err)
```

let's address this as part of https://github.com/filecoin-project/lotus/issues/12633

for this one, we just need to be sure that we pass the `ErrNullRound` on as it is - your special case branch is returning a new error that doesn't include it (`%w` would have been OK, but we also may as well just return it.

As per https://github.com/filecoin-project/lotus/issues/12633, the important part is that the text contained in the existing `ErrNullRound` is still included, regardless of what else we include, so we either return a fresh `ErrNullRound` (and perhaps decorate it further with the block number, as long as it just extends, and doesn't change, the existing message), so a `%w` does that for us.